File: PE5.cpp
//PE5.cpp
//Program by Valerie Lambert

/*  Project Euler Problem 5- - - -
 *  2520 is the smallest number that can be divided by each of the
 *  numbers from 1 to 10 without any remander.
 *  What is the smallest positive number that is evenly divisible by
 *  all of the numbers from 1 to 20?
 *  - - - - - - - - - - - - - - - -
 *
 *  This can be easily solved without use of a program (I just used
 *  a calculator the first time solving this), but a brute force
 *  approach is below. After reading the pdf for this problem I
 *  know that one can use log to find the power each term in the
 *  prime factorization sohuld be raised to. I'll use that approach
 *  next.
 */

#include<iostream>
int MAX_DIV = 20;

bool isEvenlyDivisible(int n)
{
    for (int i = 1; i < MAX_DIV; i++)
    {
        if (n%i != 0) 
        {
            return false;
        }
    }
    return true;
}

int main()
{
    int num = 1;
    while (!isEvenlyDivisible(num))
    {
        num += 1;
    }
    std::cout << num << std::endl;
    return 0;
}

File: PE5.py
#PE5.py
#Program by Valerie Lambert

"""
Project Euler Problem 5- - - -
2520 is the smallest number that can be divided by each of the numbers
from 1 to 10 without any remainder.
What is the smallest positive number that is evenly divisible by all of
the numbers from 1 to 20?
- - - - - - - - - - - - - - - -

Python version of program. Unfortunately the pure brute force method
takes a long time in python, tohugh I'm not sure why because python isn't
dealing with lists or strings... Will have to come up with a more elegant
solution.
"""

MAX_DIV = 20

def isEvenlyDivisible(num):
    for i in range(MAX_DIV):
        if num%(i+1) != 0:
            return False
    return True

def main():
    num = 2
    while not isEvenlyDivisible(num):
        num += 2
    print(num)

main()
